From 1554e50386ed86e31385c319539dd9f492f7a2da Mon Sep 17 00:00:00 2001
From: "neil.shi" <neil.shi@quectel.com>
Date: Wed, 7 Aug 2024 11:13:55 +0300
Subject: [PATCH 047/188] Revert "ath11k: do not restore ASPM in case of single
 MSI vector"

This reverts commit 915a081ff307d61d6551d6c16b542e03775353c4.

Quectel:
Enabling ASPM has some undetermined issue in one MSI vector case,
so it is disabled by default in upstream.
However in this specific project, we did a large amount of tests and confirm
it is OK to enable it. So we will not upstream it.

Signed-off-by: neil.shi <neil.shi@quectel.com>
Cherry-picked-from: 8a797047843f ("Subject: [PATCH 38/42] Revert "ath11k: do not restore ASPM in case of single MSI vector"")
Signed-off-by: Eugen Hristev <eugen.hristev@collabora.com>
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 drivers/net/wireless/ath/ath11k/pci.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/pci.c b/drivers/net/wireless/ath/ath11k/pci.c
index 83f89b8b8..3f7c69c01 100644
--- a/drivers/net/wireless/ath/ath11k/pci.c
+++ b/drivers/net/wireless/ath/ath11k/pci.c
@@ -887,14 +887,7 @@ static int ath11k_pci_start(struct ath11k_base *ab)
 {
 	struct ath11k_pci *ab_pci = ath11k_pci_priv(ab);
 
-	/* TODO: for now don't restore ASPM in case of single MSI
-	 * vector as MHI register reading in M2 causes system hang.
-	 */
-	if (test_bit(ATH11K_FLAG_MULTI_MSI_VECTORS, &ab->dev_flags))
-		ath11k_pci_aspm_restore(ab_pci);
-	else
-		ath11k_info(ab, "leaving PCI ASPM disabled to avoid MHI M2 problems\n");
-
+	ath11k_pci_aspm_restore(ab_pci);
 	ath11k_pcic_start(ab);
 
 	return 0;
-- 
2.55.0

